src/App.jsx

```diff
@@ -14,33 +15,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
+const sortFieldAlphabet = 'alphabet';
+const sortFieldLength = 'length';
+const reverseFieldGoods = 'reverse';
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [activeSort, setActiveSort] = useState('');
+  const [originalGoods, setOriginalGoods] = useState(goodsFromServer);
+
+  const reset = () => {
+    setVisibleGoods(goodsFromServer);
+    setActiveSort('');
+  };
+
+  const sortByAlphabet = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort((good1, good2) => good1.localeCompare(good2))
+    );
+    setActiveSort(sortFieldAlphabet);
+  };
+
+  const sortByLength = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort((good1, good2) => good1.length - good2.length)
+    );
+    setActiveSort(sortFieldLength);
+  }
+
+  const reverseGoods = () => {
```

Comment:
The method name 'reverseGoods' does not start with a verb. According to the checklist, the method's name should start with a verb. Consider renaming it to 'reverseOrderOfGoods'.